### PR TITLE
ome-files 0.2.2 (new formulae)

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Upstream, the bioformats-cpp project has been renamed and split into a number of smaller projects.  This PR removes bioformats-cpp and adds the separate components as individual formulae.

### CODE REVIEW

**Comment by rleigh-codelibre** [comment]
Hi folks, is there anything blocking this being merged?  Any additional information I can provide about it or anything else I can assist with?

Thanks,
Roger

File: ome-common.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  depends_on "boost"
+  depends_on "cmake" => :build
+  depends_on "xalan-c"
+  depends_on "xerces-c"
+  depends_on "doxygen" => :build if build.with? "api-docs"
+  depends_on "graphviz" => :build if build.with? "api-docs"
+
+  # Needs clang/libc++ toolchain; mountain lion is too broken
+  depends_on :macos => :mavericks
```
Comment (dpo) [inline ome-common.rb]: Does this build on Linux?

File: ome-common.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  # Required for testing
+  resource "gtest" do
+    url "https://github.com/google/googletest/archive/release-1.8.0.tar.gz"
+    sha256 "58a6f4277ca2bc8565222b3bbd58a177609e9c488e8a72649359ba51450db7d8"
+  end
+
+  def install
+    if build.with? "test"
+      (buildpath/"gtest-source").install resource("gtest")
+    end
```
Comment (dpo) [inline ome-common.rb]: You can make this a one liner:
```
(buildpath/"gtest-source").install resource("gtest") if build.with? "test"
```

File: ome-common.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    args = *std_cmake_args + [
+      "-DGTEST_ROOT=#{buildpath}/gtest",
+      "-Wno-dev",
+    ]
+
+    if build.with? "api-docs"
+      args << "-Ddoxygen:BOOL=ON"
+    else
+      args << "-Ddoxygen:BOOL=OFF"
+    end
```
Comment (dpo) [inline ome-common.rb]:
```
args << ("-Ddoxygen:BOOL=" + (build.with?("api-docs") ? "ON" : "OFF"))
```

File: ome-common.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make"
+      system "ctest", "-V"
+
+      system "make", "install"
```
Comment (dpo) [inline ome-common.rb]: Let's remove this unnecessary blank line.

File: ome-common.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      #include <ome/common/xsl/Platform.h>
+
+      int main()
+      {
+        ome::common::xml::Platform xmlplat;
+        ome::common::xsl::Platform xslplat;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-I#{include}", "-L#{lib}", "-L#{Formula["boost"].opt_lib}", "-L#{Formula["xerces-c"].opt_lib}", "-L#{Formula["xalan-c"].opt_lib}", "-pthread", "-lome-common", "-lxerces-c", "-lxalan-c", "-lboost_filesystem-mt", "-lboost_system-mt", "-o", "test"
```
Comment (dpo) [inline ome-common.rb]: Please use `opt_lib` and `opt_include` instead of `lib` and `include`.

File: ome-common.rb (hunk 2 above)
Comment (dpo) [inline ome-common.rb]: Well, sorry, is the `end` in the wrong place here?

File: ome-files.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "graphicsmagick" => :optional # For unit tests only
+  depends_on "ome-common"
+  depends_on "ome-xml"
+  depends_on "doxygen" => :build if build.with? "api-docs"
+  depends_on "graphviz" => :build if build.with? "api-docs"
+
+  # Needs clang/libc++ toolchain; mountain lion is too broken
+  depends_on :macos => :mavericks
```
Comment (dpo) [inline ome-files.rb]: Linux?

File: ome-files.rb [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    args = *std_cmake_args + [
+      "-DGTEST_ROOT=#{buildpath}/gtest",
+      "-Wno-dev",
+    ]
+
+    if build.with? "api-docs"
+      args << "-Ddoxygen:BOOL=ON"
+    else
+      args << "-Ddoxygen:BOOL=OFF"
+    end
```
Comment (dpo) [inline ome-files.rb]: Could be a one liner.

File: ome-files.rb [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+    if build.with? "docs"
+      args << "-Dsphinx:BOOL=ON"
+    else
+      args << "-Dsphinx:BOOL=OFF"
+    end
```
Comment (dpo) [inline ome-files.rb]: Ditto.

File: ome-xml.rb [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  depends_on "boost"
+  depends_on "cmake" => :build
+  depends_on "xalan-c"
+  depends_on "xerces-c"
+  depends_on "ome-common"
+  depends_on "doxygen" => :build if build.with? "api-docs"
+  depends_on "graphviz" => :build if build.with? "api-docs"
+
+  # Needs clang/libc++ toolchain; mountain lion is too broken
+  depends_on :macos => :mavericks
```
Comment (dpo) [inline ome-xml.rb]: Same comments here.

File: ome-common.rb (hunk 2 above)
Comment (rleigh-codelibre) [inline ome-common.rb]: Yes, I'll fix this.

File: ome-common.rb (hunk 1 above)
Comment (rleigh-codelibre) [inline ome-common.rb]: Yes.  I'll switch to use `MaximumMacOSRequirement` if that's the correct way to enforce the minimum version while also building on Linux?

**Comment by rleigh-codelibre** [comment]
@dpo Thanks for the review, much appreciated.  The commit I've pushed addresses all your comments to the best of my knowledge.

**Comment by rleigh-codelibre** [comment]
@dpo All the checks are now green, so I hope this is OK.  Happy to make any futher changes if needed.

**Comment by dpo** [comment]
Looks great. Many thanks for your contribution! 👍

**Pull request closed without merging**